src/input_xml.F90

```diff
@@ -4661,15 +4653,25 @@ subroutine read_mg_cross_sections_header(libraries)
     ! Open file for reading
     file_id = file_open(path_cross_sections, 'r', parallel=.true.)
 
-    if (attribute_exists(file_id, "groups")) then
-      ! Get neutron group count
-      call read_attribute(energy_groups, file_id, "groups")
+    if (attribute_exists(file_id, "energy_groups")) then
+      ! Get neutron energy group count
+      call read_attribute(num_energy_groups, file_id, "energy_groups")
     else
-      call fatal_error("'groups' attribute must exist!")
+      call fatal_error("'energy_groups' attribute must exist!")
     end if
 
-    allocate(rev_energy_bins(energy_groups + 1))
-    allocate(energy_bins(energy_groups + 1))
+    if (attribute_exists(file_id, "delayed_groups")) then
+      ! Get neutron delayed group count
+      call read_attribute(num_delayed_groups, file_id, "delayed_groups")
+    else
+      num_delayed_groups = 0
+      call write_message("WARNING: delayed_groups element not provided so &
```

Comment:
Yeah, probably best not warn about it since most people won't use it. My mindset is just so tailored towards transients now that I worry that someone would try to run a simulation but not set the delayed groups, so all the delayed neutron cross sections would be lumped integrated over delayed groups. I'll remove this warning for now.